  INFO root [0x00005acc] [2019-05-30 01:21:01,939] - 
  INFO root [0x00005acc] [2019-05-30 01:21:01,940] - -----------------------------------------------------------------------------------------


  INFO root [0x00005acc] [2019-05-30 01:21:01,944] - Resmgr::initialize: OURO_ROOT=C:/Users/Black/Ouroboros/
  INFO root [0x00005acc] [2019-05-30 01:21:01,944] - Resmgr::initialize: OURO_RES_PATH=C:/Users/Black/Ouroboros/ouro/res/;C:/Users/Black/Ouroboros/assets/;C:/Users/Black/Ouroboros/assets/scripts/;C:/Users/Black/Ouroboros/assets/res/
  INFO root [0x00005acc] [2019-05-30 01:21:01,945] - Resmgr::initialize: OURO_BIN_PATH=C:/Users/Black/Ouroboros/ouro/bin/server/
  INFO root [0x00005acc] [2019-05-30 01:21:01,959] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.27.1 ( 192.168.27.1:9859 )
  INFO root [0x00005acc] [2019-05-30 01:21:01,960] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.27.1:9859, SOMAXCONN=5.
  INFO root [0x00005acc] [2019-05-30 01:21:01,960] - server-configs:
  INFO root [0x00005acc] [2019-05-30 01:21:01,960] - 	gameUpdateHertz : 10
  INFO root [0x00005acc] [2019-05-30 01:21:01,961] - 	defaultViewRadius : 80
  INFO root [0x00005acc] [2019-05-30 01:21:01,961] - 	defaultViewHysteresisArea : 5
  INFO root [0x00005acc] [2019-05-30 01:21:01,961] - 	entryScriptFile : ouromain
  INFO root [0x00005acc] [2019-05-30 01:21:01,961] - 	internalTcpAddr : 192.168.27.1:9859
  INFO root [0x00005acc] [2019-05-30 01:21:01,961] - 	componentID : 8001
  INFO root [0x00005acc] [2019-05-30 01:21:02,011] - Components::findLogger: found logger, addr:192.168.27.1:9616
  INFO root [0x00005acc] [2019-05-30 01:21:02,012] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [0x00005acc] [2019-05-30 01:21:02,012] - Components::addComponent[logger], uid=10103, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [0x00005acc] [2019-05-30 01:21:02,019] - ---- cellapp Version: 2.5.0. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: 8652043B9D1A1F169DDC4E8CD899D692. Config: Debug X86. Built: 02:55:12 May 23 2019. AppID: 8001. UID: 10103. PID: 26844 ----
  INFO root [0x00005acc] [2019-05-30 01:21:02,020] - Server Black: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz with 2097151kB RAM
  INFO root [0x00005acc] [2019-05-30 01:21:02,021] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [0x00005acc] [2019-05-30 01:21:02,022] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [0x00005acc] [2019-05-30 01:21:02,222] - Script::install(): is successfully, Python=(3.7.3 (default, May 23 2019, 03:58:29) [MSC v.1916 32 bit (Intel)])!
  INFO root [0x00005acc] [2019-05-30 01:21:02,222] - Script::install(): OUROxtra is successfully!
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,556] - loaded entity-script:Account(1).
  WARN root [0x00005acc] [2019-05-30 01:21:02,558] - EntityDef::checkDefMethod: exposed of method: Avatar.updateEG, check the caller can use "def updateEG(self, callerID, ...)", such as: if callerID == self.id!
  WARN root [0x00005acc] [2019-05-30 01:21:02,558] - EntityDef::checkDefMethod: exposed of method: Avatar.updateLastSyncTime, check the caller can use "def updateLastSyncTime(self, callerID, ...)", such as: if callerID == self.id!
  WARN root [0x00005acc] [2019-05-30 01:21:02,559] - EntityDef::checkDefMethod: exposed of method: Avatar.updatePositionSynced, check the caller can use "def updatePositionSynced(self, callerID, ...)", such as: if callerID == self.id!
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,559] - loaded entity-script:Avatar(2).
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,560] - loaded entity-script:FirstEntity(3).
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,560] - loaded entity-script:Scene(4).
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,560] - loaded entity-script:Space(5).
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,561] - loaded entity-script:SpaceDuplicate(6).
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,562] - loaded entity-script:SpawnPoint(8).
  INFO root [0x00005acc] [2019-05-30 01:21:02,563] - TelnetServer server is running on port 50000
  INFO root [0x00005acc] [2019-05-30 01:21:02,990] - ---- cellapp is running ----
 DEBUG root [0x00005acc] [2019-05-30 01:21:02,990] - Components::process(): Request for the process of identity...
  WARN root [0x00005acc] [2019-05-30 01:21:09,263] - Forwarding logs to logger[192.168.27.1:9616/0/0/0]...
 ERROR root [0x00005acc] [2019-05-30 04:00:39,228] - Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220992155048, lastReceivedTime=220953101719, inactivityExceptionPeriod=599500512)
  INFO root [0x00005acc] [2019-05-30 04:00:39,237] - The following logs sent to logger failed:
  WARN root [0x00005acc] [2019-05-30 04:00:39,238] - ==> [2019-05-30 04:00:39 228] TCPPacketReceiver::processPendingEvents(192.168.27.1:9616): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [0x00005acc] [2019-05-30 04:00:39,238] - ==> [2019-05-30 04:00:39 229] Components::removeComponentByChannel: logger : 2000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220992155048, lastReceivedTime=220953101719, inactivityExceptionPeriod=599500512)
  WARN root [0x00005acc] [2019-05-30 04:00:39,355] - TCPPacketReceiver::processPendingEvents(192.168.27.1:9696): Throwing REASON_GENERAL_NETWORK - WSAECONNRESET
 ERROR root [0x00005acc] [2019-05-30 04:00:39,357] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=220993444919, lastReceivedTime=220990922844, inactivityExceptionPeriod=599500512)
  INFO root [0x00005acc] [2019-05-30 04:00:39,362] - Shutdowner::onShutdownBegin: shutting down(period=0, tickPeriod=1)
  INFO root [0x00005acc] [2019-05-30 04:00:39,363] - Shutdowner::onShutdown: shutting down
  INFO root [0x00005acc] [2019-05-30 04:00:39,363] - Spaces::destroySpace: 1.
  INFO root [0x00005acc] [2019-05-30 04:00:39,364] - Spaces::destroySpace: 2.
  INFO root [0x00005acc] [2019-05-30 04:00:39,365] - Shutdowner::onShutdownEnd: shutting down
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,366] - Space::~Space(): 3
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,367] - Space::~Space(): 2
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,367] - NavMeshHandle::~NavMeshHandle(): (spaces/novice_1) is destroyed!
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,367] - NavMeshHandle::~NavMeshHandle(): (spaces/novice_2) is destroyed!
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,368] - EntityRef::destroyObjPool(): size 16.
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,368] - Witness::destroyObjPool(): size 16.
 ERROR root [0x00005acc] [2019-05-30 04:00:39,382] - PyGC::debugTracing(): EntityCall : leaked(1)
  INFO root [0x00005acc] [2019-05-30 04:00:39,402] - Script::uninstall(): is successfully!
 DEBUG root [0x00005acc] [2019-05-30 04:00:39,402] - ThreadPool::destroy(): starting size 2.
  WARN root [0x00005acc] [2019-05-30 04:00:39,703] - ThreadPool::destroy(): waiting for thread(2)[0xb73d38,0xb74488,], try=1
  INFO root [0x00001cf8] [2019-05-30 04:01:02,487] - 
  INFO root [0x00001cf8] [2019-05-30 04:01:02,488] - -----------------------------------------------------------------------------------------


  INFO root [0x00001cf8] [2019-05-30 04:01:02,489] - Resmgr::initialize: OURO_ROOT=C:/Users/Black/Ouroboros/
  INFO root [0x00001cf8] [2019-05-30 04:01:02,489] - Resmgr::initialize: OURO_RES_PATH=C:/Users/Black/Ouroboros/ouro/res/;C:/Users/Black/Ouroboros/assets/;C:/Users/Black/Ouroboros/assets/scripts/;C:/Users/Black/Ouroboros/assets/res/
  INFO root [0x00001cf8] [2019-05-30 04:01:02,490] - Resmgr::initialize: OURO_BIN_PATH=C:/Users/Black/Ouroboros/ouro/bin/server/
  INFO root [0x00001cf8] [2019-05-30 04:01:02,506] - NetworkInterface::initialize(INTERNAL-TCP): bound to all interfaces with default route interface on 192.168.27.1 ( 192.168.27.1:1950 )
  INFO root [0x00001cf8] [2019-05-30 04:01:02,507] - NetworkInterface::initialize(INTERNAL-TCP): address 192.168.27.1:1950, SOMAXCONN=5.
  INFO root [0x00001cf8] [2019-05-30 04:01:02,508] - server-configs:
  INFO root [0x00001cf8] [2019-05-30 04:01:02,508] - 	gameUpdateHertz : 10
  INFO root [0x00001cf8] [2019-05-30 04:01:02,509] - 	defaultViewRadius : 80
  INFO root [0x00001cf8] [2019-05-30 04:01:02,509] - 	defaultViewHysteresisArea : 5
  INFO root [0x00001cf8] [2019-05-30 04:01:02,509] - 	entryScriptFile : ouromain
  INFO root [0x00001cf8] [2019-05-30 04:01:02,510] - 	internalTcpAddr : 192.168.27.1:1950
  INFO root [0x00001cf8] [2019-05-30 04:01:02,510] - 	componentID : 8001
  INFO root [0x00001cf8] [2019-05-30 04:01:03,063] - Components::findLogger: found logger, addr:192.168.27.1:1945
  INFO root [0x00001cf8] [2019-05-30 04:01:03,064] - Components::addComponent: reset orderLog, uid=10103!
  INFO root [0x00001cf8] [2019-05-30 04:01:03,064] - Components::addComponent[logger], uid=10103, componentID=2000, globalorderid=1, grouporderid=1, totalcount=1
  INFO root [0x00001cf8] [2019-05-30 04:01:03,070] - ---- cellapp Version: 2.5.0. ScriptVersion: 0.1.0. Pythoncore: 3.7.3. Protocol: 8652043B9D1A1F169DDC4E8CD899D692. Config: Debug X86. Built: 02:55:12 May 23 2019. AppID: 8001. UID: 10103. PID: 26128 ----
  INFO root [0x00001cf8] [2019-05-30 04:01:03,070] - Server Black: Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz with 2097151kB RAM
  INFO root [0x00001cf8] [2019-05-30 04:01:03,071] - ThreadPool::createThreadPool: creating  threadpool...
  INFO root [0x00001cf8] [2019-05-30 04:01:03,071] - ThreadPool::createThreadPool: successfully(2), newThreadCount=1, normalMaxThreadCount=2, maxThreadCount=8
  INFO root [0x00001cf8] [2019-05-30 04:01:03,282] - Script::install(): is successfully, Python=(3.7.3 (default, May 23 2019, 03:58:29) [MSC v.1916 32 bit (Intel)])!
  INFO root [0x00001cf8] [2019-05-30 04:01:03,283] - Script::install(): OUROxtra is successfully!
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,603] - loaded entity-script:Account(1).
  WARN root [0x00001cf8] [2019-05-30 04:01:03,605] - EntityDef::checkDefMethod: exposed of method: Avatar.updateEG, check the caller can use "def updateEG(self, callerID, ...)", such as: if callerID == self.id!
  WARN root [0x00001cf8] [2019-05-30 04:01:03,605] - EntityDef::checkDefMethod: exposed of method: Avatar.updateLastSyncTime, check the caller can use "def updateLastSyncTime(self, callerID, ...)", such as: if callerID == self.id!
  WARN root [0x00001cf8] [2019-05-30 04:01:03,606] - EntityDef::checkDefMethod: exposed of method: Avatar.updatePositionSynced, check the caller can use "def updatePositionSynced(self, callerID, ...)", such as: if callerID == self.id!
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,606] - loaded entity-script:Avatar(2).
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,607] - loaded entity-script:FirstEntity(3).
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,607] - loaded entity-script:Scene(4).
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,608] - loaded entity-script:Space(5).
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,608] - loaded entity-script:SpaceDuplicate(6).
 DEBUG root [0x00001cf8] [2019-05-30 04:01:03,610] - loaded entity-script:SpawnPoint(8).
  INFO root [0x00001cf8] [2019-05-30 04:01:03,611] - TelnetServer server is running on port 50000
  INFO root [0x00001cf8] [2019-05-30 04:01:04,092] - ---- cellapp is running ----
 DEBUG root [0x00001cf8] [2019-05-30 04:01:04,092] - Components::process(): Request for the process of identity...
  WARN root [0x00001cf8] [2019-05-30 04:01:10,428] - Forwarding logs to logger[192.168.27.1:1945/0/0/0]...
 ERROR root [0x00001cf8] [2019-05-30 05:40:20,788] - Components::removeComponentByChannel: dbmgr : 4000, Abnormal exit(reason=TCPPacketReceiver::processRecv(): error=10054
)! Channel(timestamp=280807761522, lastReceivedTime=280621632946, inactivityExceptionPeriod=599500064)
 ERROR root [0x00001cf8] [2019-05-30 05:40:20,816] - 192.168.27.1:1951: TCPPacketSender::processSend(internal): Exception occurred: REASON_CLIENT_DISCONNECTED , errno: 10054, 192.168.27.1:1951/0/0/0
 ERROR root [0x00001cf8] [2019-05-30 05:40:20,819] - 192.168.27.1:1949: TCPPacketSender::processSend(internal): Exception occurred: REASON_CLIENT_DISCONNECTED , errno: 10054, 192.168.27.1:1949/0/0/0
 ERROR root [0x00001cf8] [2019-05-30 05:40:20,844] - PyGC::debugTracing(): EntityCall : leaked(1)
